src/main/java/dev/morling/onebrc/CalculateAverage_yasyf.java

```diff
@@ -0,0 +1,262 @@
+/*
+ *  Copyright 2023 The original authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package dev.morling.onebrc;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.lang.foreign.Arena;
+import java.lang.foreign.MemorySegment;
+import java.lang.reflect.Field;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileChannel.MapMode;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.TreeMap;
+
+import sun.misc.Unsafe;
+
+public class CalculateAverage_yasyf {
+    private static final int CORES = Runtime.getRuntime().availableProcessors() * 2;
+    private static final int N_CITIES = 1000;
```

Comment:
> There is a maximum of 10,000 unique station names

Doesn't fit the rules, if that constant is the limit of unique cities.

Reply:
fixed! thx